ee/clickhouse/models/event.py

```diff
@@ -96,22 +97,49 @@ def bulk_create_events(events: List[Dict[str, Any]], person_mapping: Optional[Di
         )
 
         #  use person properties mapping to populate person properties in given event
+        team_id = event["team"].pk if event.get("team") else event["team_id"]
         if person_mapping and person_mapping.get(event["distinct_id"]):
-            person_properties = person_mapping[event["distinct_id"]].properties
-            person_id = person_mapping[event["distinct_id"]].uuid
+            person = person_mapping[event["distinct_id"]]
+            person_properties = person.properties
+            person_id = person.uuid
+        else:
+            try:
+                person = Person.objects.get(
+                    persondistinctid__distinct_id=event["distinct_id"], persondistinctid__team_id=team_id
+                )
+                person_properties = person.properties
+                person_id = person.uuid
+            except Person.DoesNotExist:
+                person_properties = {}
+                person_id = cast(uuid.UUID, "00000000-0000-0000-0000-000000000000")
```

Comment:
something interesting I found in funnels PR that may need to be accounted for is that null personids could end up being counted in unexpected ways. For example, on trends, if we're looking at unique users (DISTINCT events.person_id), all anonymous users will get collapsed into the same user

Reply:
Do anonymous users not have personIDs? 🤔

Reply:
Seems like we're good here: https://posthog.slack.com/archives/C03E27SM90S/p1653059142798669

Need to fix this though - no case for person_id being `0000...`.